src/Functions/FunctionsConversion.cpp

```diff
@@ -1424,7 +1426,11 @@ struct ConvertImpl
                             else
                                 throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Provided time zone must be non-empty");
                         }
-                        bool is_ok = FormatImpl<FromDataType>::template execute<bool>(vec_from[i], write_buffer, &type, time_zone);
+                        bool is_ok = true;
+                        if constexpr (std::is_same_v<FromDataType, DataTypeDateTime64>)
+                            writeDateTimeText(DateTime64(vec_from[i]), type.getScale(), write_buffer, *time_zone, trim_suffix_zeros);
```

Comment:
better add a ut about `writeDateTimeText` in `src/IO/tests`

Reply:
Thanks, I will.